src/binding_utils.rs

```diff
@@ -135,6 +136,11 @@ pub fn rs_to_py_regex<'py>(py: Python<'py>, regex: &Regex) -> PyResult<&'py PyAn
     re.call_method1("compile", args)
 }
 
+pub fn to_trustchain_error_exception(err: libparsec::core::TrustchainError) -> PyErr {
+    let python_error = TrustchainError(err);
+    TrustchainErrorException::new_err(python_error)
+}
```

Comment:
Using this code block into the trait `impl From<libparsec::core::TrustchainError> for PyErr` would simplify the error handling part

as we don't need to use `.map_err(to_trustchain_error_exception)` before unwrapping with `?` when we use rust code the return `libparsec::core::TrustchainError`

Reply:
I agree but shouldn't we avoid to put python related code in `libparsec::core` ? We can't implement `From<libparsec::core::TrustchainError>` in `src/trustchain.rs`

Reply:
Maybe add a file `error.rs` in `src/` ?

Reply:
It will not work we would get an error like this `only traits defined in the current crate can be implemented for types defined outside of the crate` by the compiler

Reply:
I've implemented `From<TrustchainError>` trait for `PyErr` this way we can just do a `.map_err(TrustchainError)?`

Reply:
I was thinking of something like that 

```rust
impl From<libparsec::TrustchainError> for TrustchainError {}
impl From<TrustchainError> for PyErr {}
```

Reply:
Does it matter since we'll need to do at least a `map_err` to be able to use `?`?